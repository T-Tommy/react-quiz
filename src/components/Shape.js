import React from 'react';
import PropTypes from 'prop-types';

function Shape({ text, color, background }) {
  const styles = {
    color,
    background,
    'box-sizing': 'border-box',
    display: 'inline-block',
    margin: '1em',
    padding: '1em 2em',
    'max-width': '90vw',
    'font-size': '2rem',
    'overflow-wrap': 'break-word',
    'font-family': 'Verdana, sans-serif',
    'text-align': 'center'
  };
  return (
    <div style={ styles }>
      <strong>{ text }</strong>
    </div>
  );
}

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  background: PropTypes.string.isRequired
};

export default Shape;
