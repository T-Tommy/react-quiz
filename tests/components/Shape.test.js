import React from 'react';
import { shallow } from 'enzyme';
import Shape from '../../src/components/Shape';

describe('Shape component', () => {
  it('renders the Shape', () => {
    const wrapper = shallow(<Shape text="poop" color="#AAAAAA" background="#111111" />);
    expect(wrapper).toMatchSnapshot();
  });
});
