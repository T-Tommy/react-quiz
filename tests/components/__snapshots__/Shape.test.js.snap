// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders the Shape 1`] = `
<div
  style={
    Object {
      "background": "#111111",
      "box-sizing": "border-box",
      "color": "#AAAAAA",
      "display": "inline-block",
      "font-family": "Verdana, sans-serif",
      "font-size": "2rem",
      "margin": "1em",
      "max-width": "90vw",
      "overflow-wrap": "break-word",
      "padding": "1em 2em",
      "text-align": "center",
    }
  }
>
  <strong>
    poop
  </strong>
</div>
`;
